src/sagas/reown.js

```diff
@@ -0,0 +1,996 @@
+        const sendNanoContractTxResponseTemplate = (accepted) => (data) => {
+          dispatch(hideGlobalModal());
+          resolve({
+            type: TriggerResponseTypes.SendNanoContractTxConfirmationResponse,
+            data: {
+              accepted,
+              nc: data
+            }
+          });
+        };
+
+        dispatch({
+          type: types.SHOW_NANO_CONTRACT_SEND_TX_MODAL,
+          payload: {
+            accept: sendNanoContractTxResponseTemplate(true),
+            deny: sendNanoContractTxResponseTemplate(false),
+            data: request.data,
+            dapp: requestMetadata,
+          }
+        });
+      } break;
+
+      case TriggerTypes.CreateTokenConfirmationPrompt: {
+        const createTokenResponseTemplate = (accepted) => (data) => {
+          dispatch(hideGlobalModal());
+          resolve({
+            type: TriggerResponseTypes.CreateTokenConfirmationResponse,
+            data: {
+              accepted,
+              token: data?.payload,
+            }
+          });
+        };
+
+        dispatch({
+          type: types.SHOW_CREATE_TOKEN_REQUEST_MODAL,
+          payload: {
+            accept: createTokenResponseTemplate(true),
+            deny: createTokenResponseTemplate(false),
+            data: request.data,
+            dapp: requestMetadata,
+          }
+        });
+      } break;
+
+      case TriggerTypes.SignMessageWithAddressConfirmationPrompt: {
+        const signMessageResponseTemplate = (accepted) => () => {
+          dispatch(hideGlobalModal());
+          resolve({
+            type: TriggerResponseTypes.SignMessageWithAddressConfirmationResponse,
+            data: accepted,
+          });
+        };
+
+        dispatch({
+          type: types.SHOW_SIGN_MESSAGE_REQUEST_MODAL,
+          payload: {
+            accept: signMessageResponseTemplate(true),
+            deny: signMessageResponseTemplate(false),
+            data: request.data,
+            dapp: requestMetadata,
+          }
+        });
+      } break;
+
+      case TriggerTypes.SendNanoContractTxLoadingTrigger:
+        dispatch(setNewNanoContractStatusLoading());
+        dispatch(showGlobalModal(MODAL_TYPES.NANO_CONTRACT_FEEDBACK, { isLoading: true }));
+        resolve();
+        break;
+
+      case TriggerTypes.CreateTokenLoadingTrigger:
+        log.debug('CreateTokenLoadingTrigger: Starting token creation process');
+        dispatch(setCreateTokenStatusLoading());
+        dispatch(showGlobalModal(MODAL_TYPES.TOKEN_CREATION_FEEDBACK, { isLoading: true }));
+        resolve();
+        break;
+
+      case TriggerTypes.CreateTokenLoadingFinishedTrigger:
+        log.debug('CreateTokenLoadingFinishedTrigger: Token creation process completed');
+        dispatch(setCreateTokenStatusReady());
+        dispatch(hideGlobalModal());
+        resolve();
+        break;
+
+      case TriggerTypes.SendNanoContractTxLoadingFinishedTrigger:
+        dispatch(setNewNanoContractStatusReady());
```

Comment:
Why don't you need to call `dispatch(hideGlobalModal());` like in the case above?

Reply:
We do actually need to dispatch it, fixed